src/repo.rs

```diff
@@ -1231,6 +1231,56 @@ impl Repository {
     /// current branch and make it point to this commit. If the reference
     /// doesn't exist yet, it will be created. If it does exist, the first
     /// parent must be the tip of this branch.
+    pub fn new_commit<'a, C: AsRef<Commit<'a>>>(
```

Comment:
can we add this `&[] as &[Commit<'_>]` for the empty list case to the documentation as this will be a little unintuitive 🙈

Reply:
Sure, I'll add that in. It's really too bad the default type parameter fallback feature didn't make it into the language since this issue was created.

Reply:
I think this may have actually been one of the issues which led me to the original signature, if this otherwise requires type annotations then I don't think that this is the best way to shape this API. Another option would be perhaps a builder-like construct but that's pretty wordy for just the type inference issue here.

Reply:
So there are two options here: 

1. We can stick with this, which allows existing calls that contain values to be the same (appears to be the majority of calls, based on some seen usage of this method in the wild and in the tests), but does have this edge case that Rust doesn't handle very gracefully, while supporting owned values in the slice/vector parameter. 
2. We could make a hard switch to using owned values, so the signature would be `&[Commit]` in that case. We lose compatibility with existing calls, but this edge case type inference issue goes away. 

I personally think that supporting owned values makes sense here, thus the reason for the PR, but would like to support the existing usage where possible. This makes the former seem like a good compromise to me since it only comes up in this one case, but I could go either way. 

Thoughts?